packages/Adapters/POD_C/test/tstC_API.cpp

```diff
@@ -0,0 +1,92 @@
+#include "DTK_C_API.h"
+
+#include "Teuchos_UnitTestHarness.hpp"
+#include "Teuchos_DefaultComm.hpp"
+
+#include "mpi.h"
+
+namespace {
+
+TEUCHOS_UNIT_TEST( C_API, create_apply_delete_map )
+{
+  // get communicator
+  Teuchos::RCP<const Teuchos::Comm<int>> comm =
+    Teuchos::DefaultComm<int>::getComm();
+
+  // let us use INFO format
+  std::string options;
+  options = "\n"\
+    "\"Map Type\" \"Spline Interpolation\"\n"\
+    "\"Basis Type\" \"Wendland\"\n"\
+    "\"Basis Order\" 0\n"\
+    "\"Spatial Dimension\" 3\n"\
+    "\"Target Coord Data Layout\" \"Contiguous\"\n"\
+    "\"Source Coord Data Layout\" \"Contiguous\"\n"\
+    "\"RBF Radius\" 0.1\n"\
+    "\n";
+
+  // TODO: fill these
+  std::srand(0);
+  int const space_dim = 3;
+  int const field_dim = 1;
+  unsigned const src_num = 100;
+  std::vector<double> src_coord(space_dim*src_num);
+  std::vector<double> src_field(field_dim*src_num);
+  for (unsigned i; i < src_num; ++i)
+  {
+    for (unsigned j; j < space_dim; ++j)
+    {
+      // assume data layout is contiguous
+      src_coord[i+j*src_num] = std::rand() / RAND_MAX;
+    }
+    for (unsigned j; j < field_dim; ++j)
+    {
+      // assume data layout is interleaved
+      src_field[i*field_dim+j] = static_cast<double>(j);
+    }
+  }
+  unsigned const tgt_num = 50;
+  std::vector<double> tgt_coord(space_dim*tgt_num);
+  std::vector<double> tgt_field(field_dim*tgt_num);
+  for (unsigned i; i < tgt_num; ++i)
+  {
+    for (unsigned j; j < space_dim; ++j)
+    {
+      tgt_coord[i+j*tgt_num] = std::rand() / RAND_MAX;
+    }
+    for (unsigned j; j < field_dim; ++j)
+    {
+      tgt_field[i*field_dim+j] = static_cast<double>(j);
+    }
+  }
+  TEST_EQUALITY( src_coord.size(), space_dim * src_num );
+  TEST_EQUALITY( tgt_coord.size(), space_dim * tgt_num );
+
+  DTK_Map* dtk_map = DTK_Map_create( MPI_COMM_WORLD,
+                                     src_coord.data(),
+                                     src_num,
+                                     DTK_CONTIGUOUS,
+                                     tgt_coord.data(),
+                                     tgt_num,
+                                     DTK_CONTIGUOUS,
+                                     space_dim,
+                                     options.c_str() );
+
+  options = "\n"\
+    "\"Source Field Data Layout\" \"Interleave\"\n"\
+    "\"Target Field Data Layout\" \"Interleave\"\n"\
+    "\"Field Dimension\" 1\n"\
+    "\n";
```

Comment:
Same comment here on options. What options might we need on apply? Do we need any at all?